### PR TITLE
Added toArray()

### PR DESCRIPTION
**Pull request by IvanJov**
Hi! I was using Twix on previous project and I think that's pretty useful to have `twix.toArray()` helper function.

If you like it I am happy to write some text about it in documentation. Usage is very simple, it has 4 arguments, first one is format of the moment object that will be in array, other 3 are same as `iterate` function.

Example: `moment.utc('1982-05-25').twix(moment.utc('1982-05-27'), allDay: true).toArray('YYYY-MM-DD', 'days')`
Output: `['1982-05-25', '1982-05-26', '1982-05-27']`

### CODE REVIEW

**Comment by icambron** [comment]
I like the `toArray()` idea but it shouldn't be responsible for the formatting; it should just return times. You can do this:

```js
t.toArray().map((m) -> m.format("YYYY-MM-DD"))
```

Or perhaps it can have a second optional `mapper` argument like this:

```js
t.toArray((m) -> m.format("YYYY-MM-DD"))
```

**Comment by IvanJov** [comment]
@icambron You are right, I will remove the formatting! And I don't like the idea with `mapper`argument, maybe that's too much also :)

What do you think?

**Comment by IvanJov** [comment]
Let me know if this looks better!

File: src/twix.coffee [hunk 1]
```diff
@@ -197,6 +197,13 @@ makeTwix = (moment) ->
     # -- FORMATING --
     toString: -> "{start: #{@_start.format()}, end: #{@_end.format()}, allDay: #{if @allDay then 'true' else 'false'}}"
 
+    toArray: (intervalAmount, period, minHours) ->
+      itr = @iterate(intervalAmount, period, minHours)
+      range = []
+      while itr.hasNext()
+        range.push itr.next()
+      return range
```
Comment (icambron) [inline src/twix.coffee]: style: elide `return`

File: test/twix.spec.coffee [hunk 2]
```diff
@@ -1557,6 +1558,11 @@ test = (moment, Twix) ->
       stringed = moment.utc('1982-05-25').twix(moment.utc('1982-05-25'), allDay: true).toString()
       assertEqual '{start: 1982-05-25T00:00:00Z, end: 1982-05-25T00:00:00Z, allDay: true}', stringed
 
+  describe 'toArray', ->
+    it 'returns an array of moment objects', ->
+      arrayOfDays = moment.utc('1982-05-25').twix(moment.utc('1982-05-27'), allDay: true).toArray('days').map((m) -> moment(m).format('YYYY-MM-DD'))
```
Comment (icambron) [inline test/twix.spec.coffee]: You shouldn't need the `moment(m)` since `m` is already a `Moment`

**Comment by icambron** [comment]
I left a couple of specific comments. If you fix those and also edit https://github.com/icambron/twix.js/blob/master/docs/docs.md, I'll merge this. Thanks for the contribution!

**Comment by IvanJov** [comment]
@icambron I've fixed those issues, sorry! Not so experienced with CoffeeScript, this is basically my first code written in it 😄 
And re docs, what do you think, what's the best place to put `toArray` function?

**Comment by icambron** [comment]
You know, I'll do the docs, so don't worry about that.

**Comment by IvanJov** [comment]
@icambron Ok sure, thanks! I am glad I helped! If I come to something that can be useful I'll commit again!

**Comment by icambron** [comment]
@IvanJov Actually now, that I sit down to document it, is this really different from `split`?

```js
> moment.utc('1982-05-25').twix(moment.utc('1982-05-27'), true).split(1, 'day').map(function(t){return t.start().format('YYYY-MM-DD');});
[ '1982-05-25', '1982-05-26', '1982-05-27' ]
```

**Comment by IvanJov** [comment]
@icambron Hmm, don't know, didn't think about `split`. I think that with `toArray` it's kinda easier to put range into array.

**Comment by icambron** [comment]
I'm going to leave it in. Releasing as 1.1.2

**Pull request merged**